### PR TITLE
Workflow for trash-bucket (activate/deactivate) projects.

### LINKED ISSUE #454: Trash management for projects
### Checklist

- [X] I've searched through the [current issues](https://github.com/g3w-suite/g3w-admin/issues) to make sure this feature hasn't been requested already.

### Motivation

To prevent a project from being permanently deleted, it would be helpful to have a delete flow that includes an intermediate step such as a *trash* can.

### Suggested solution

- Add a flag to projects table to make a project into *trash state*.
- Create a *trash page* to show and permanently delete or restore a project.

### Alternatives considered

_No response_

### PR DESCRIPTION
**Pull request by wlorenzetti**
Closes: #454

### CODE REVIEW

**Comment by Raruto** [comment]
### Preface

I don't quite understand why, since I'm working with a local version of g3w-admin (ref: `G3WSUITE_LOCAL_CODE_PATH=../g3w-admin`) but I had to update the docker image to make this dependency available, ref: https://github.com/g3w-suite/g3w-admin/pull/451 (ie. deleting the `setup_done` or `build_done` file didn't change anything).

## Test perfomed: "migrating a previous installation"

<details>

<summary> Click here for more info </summary>

Starting from this commit: https://github.com/g3w-suite/g3w-admin/pull/456/commits/862a0c5b329697bcba6b6702cf8af9e0f3ba504b, tested my local installation on this branch I get and I get that all projects are trashed:

image

image

</details>

## Test performed: "disallow logged out users from accessing deleted groups"

When all groups are trashed, the anonymous user should not have access to any of these groups (and projects, or at least I think so):

image

image

**Comment by wlorenzetti** [comment]
@Raruto maybe you meant me :) not @volterra79 .
Every projects and  groups of projects are the property `is_active` on every groups and projects is False, the property is used for check if a whether an item can be considered trashed.

I'm working now to set the property `is_active` fro groups and projects to True by model migrations.
If you wanto to try the PR you have to set is_active to True on every projects and groups inside your db.

**Comment by Raruto** [comment]
> @Raruto maybe you meant me :) not @volterra79 .

Damn accurate Wolter of Wolterra .. 🐺

> Every projects and groups of projects are the property `is_active` on every groups and projects is `False`, the property is used for check if a whether an item can be considered trashed.

> I'm working now to set the property is_active fro groups and projects to True by model migrations.
If you want to try the PR you have to set `is_active` to `True` on every projects and groups inside your db.

Ok, thanks for the info.

For the moment I would focus just on the starting situation (without further modifications on the db). In this sense I edited the comment above by adding another simple test: https://github.com/g3w-suite/g3w-admin/pull/456#issuecomment-1443536601.

👋 Raruto

**Comment by wlorenzetti** [comment]
Sure @Raruto, this feature is for next v3.6.x release, after that it'll be necessary to do an upgrade of every third-part module and plugins for g3w-suite core application.

File: g3w-admin/core/sitetrees.py [hunk 1]
```diff
@@ -34,6 +36,7 @@
         item('Cartographic groups', 'group-list', icon_css_class='fa fa-globe', children=[
           item('Add group', 'group-add', url_as_pattern=True, icon_css_class='fa fa-plus', access_by_perms=['core.add_group']),
           item('Groups list', 'group-list', url_as_pattern=True, icon_css_class='fa fa-globe', alias='group-list', in_breadcrumbs=False),
+          item('Bucket', 'group-deactive-list', url_as_pattern=True, icon_css_class='fa fa-trash', alias='group-deactive-list', in_breadcrumbs=True),
```
Comment (Raruto) [inline g3w-admin/core/sitetrees.py]: Perhaps `Trash` or `Recycle bin` may be more appropriate?

image

The other one reminds so much: [AWS S3](https://aws.amazon.com/en/s3/) buckets

Comment (wlorenzetti) [reply g3w-admin/core/sitetrees.py]: Yes thanks.

**Review by Raruto — COMMENTED** [review]
> Sure @Raruto, this feature is for next v3.6.x release, after that it'll be necessary to do an upgrade of every third-part module and plugins for g3w-suite core application.

Definitely, with https://github.com/g3w-suite/g3w-admin/pull/456#issuecomment-1443536601 I just meant to point out that deactivated projects should not be included by default in sql queries (or rest api requests) unless you are running queries that specifically include them in search results.

👋 Raruto

**Comment by wlorenzetti** [comment]
> Definitely, with [#456 (comment)](https://github.com/g3w-suite/g3w-admin/pull/456#issuecomment-1443536601) I just meant to point out that deactivated projects should not be included by default in sql queries (or rest api requests) unless you are running queries that specifically include them in search results.

yes thanks, I didn't think about serch results, other situations I think I've done it

**Pull request merged**